### PR TITLE
Change ProtoAdapter to use KClass instead of java.lang.Class

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

**Comment by swankjesse** [comment]
Impact on Java callers?

**Comment by Egorand** [comment]
`ProtoAdapter` and `FieldEncoding` constructors accepting `java.lang.Class` are gone, so Java callers will have to pass in a `KClass`, which can be obtained using `JvmMappingsKt.getKotlinClass(java.lang.Class)` - see regenerated tests for an example. Since runtime depends on the stdlib anyway, it should just be a minor annoyance.

File: wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.kt [hunk 1]
```diff
@@ -115,7 +115,7 @@ internal class MessageTypeAdapter<M : Message<M, B>, B : Message.Builder<M, B>>(
       if (element.isJsonNull) {
         return emptyList<Any>()
       }
-      val itemType = fieldBinding.singleAdapter().javaType
+      val itemType = fieldBinding.singleAdapter().type?.javaObjectType
```
Comment (JakeWharton) [inline wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.kt]: Why are these all using safe-call?

File: wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/RuntimeMessageAdapter.kt [hunk 2]
```diff
@@ -62,9 +63,11 @@ class RuntimeMessageAdapter<M : Message<M, B>, B : Builder<M, B>>(
     for (fieldBinding in fieldBindings.values) {
       if (fieldBinding.redacted && fieldBinding.label == WireField.Label.REQUIRED) {
         throw UnsupportedOperationException(
-            "Field '${fieldBinding.name}' in ${javaType?.name} is required and cannot be redacted.")
+            "Field '${fieldBinding.name}' in ${type?.javaObjectType?.name} is required and " +
+                "cannot be redacted.")
       }
-      val isMessage = Message::class.java.isAssignableFrom(fieldBinding.singleAdapter().javaType)
+      val isMessage = Message::class.java
+          .isAssignableFrom(fieldBinding.singleAdapter().type?.javaObjectType)
```
Comment (JakeWharton) [inline wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/RuntimeMessageAdapter.kt]: Should this be `!!`

**Pull request merged**